sgv2-dynamo-api/src/main/java/io/stargate/sgv2/dynamosvc/resources/DynamoResource.java

```diff
@@ -85,17 +92,24 @@ public Response handleRequest(
           GetItemRequest getItemRequest = awsRequestMapper.readValue(payload, GetItemRequest.class);
           result = itemProxy.getItem(getItemRequest, bridge);
           break;
+        case Query:
+          QueryRequest queryRequest = awsRequestMapper.readValue(payload, QueryRequest.class);
+          result = queryProxy.query(queryRequest, bridge);
+          break;
         default:
           throw new WebApplicationException(
               "Invalid statement type: " + target, Response.Status.BAD_REQUEST);
       }
       response = awsRequestMapper.writeValueAsBytes(result);
     } catch (JsonProcessingException ex) {
+      logger.info("Invalid payload", ex);
```

Comment:
Typically one should either throw an exception further OR log, but not both? In this case, specifically, logs could be full of "Invalid payload" with no useful information (request id, caller, etc etc). Then again, exception message itself does not have much more information.

Of course there may be value in logging something about invalid requests beyond metrics so sometimes logging makes sense too.